lib/shopify/multipass.ex

```diff
@@ -0,0 +1,114 @@
+  # Refer to the erlang docs as to which block sizes are allowed
+  @block_size 16
+
+  @doc ~S"""
+  Simple getter to expose the module attribute
+  """
+  def get_block_size(), do: @block_size
+
+  @doc ~S"""
+  Get the full URL for the Multipass signin.
+
+  ## Examples
+    iex> customer_data = %{
+    ...>     email: "something@test.shopify.com",
+    ...>     created_at: "2019-03-07T20:29:42.177390Z"
+    ...>   }
+    %{
+    created_at: "2019-03-07T20:29:42.177390Z",
+    email: "something@test.shopify.com"
+    }
+    iex> secret = "1234567890abcdef1234567890abcdef"
+    "1234567890abcdef1234567890abcdef"
+    iex>  url = Shopify.Multipass.get_url("myteststore", customer_data, secret)
+    "https://myteststore.myshopify.com/account/login/multipass/moaqEVx1Yu9hsvYvVpj-LeRYDtOo6ikicfTZd8tR8-xBMRg8tFjGEfllEcjj2VdbsezmT0XuEdglyQzi_biQPkfLJnP1dkxhNtfzwtt6IMQzu3W0qCPzbrUMD_gLaytPVP-zZZuYiSBqEMNdvzFg3zf0TOQHwbizX2D7It02sFI7ZpTRhfX4m_crV0b-DmmF"
+    iex>  url = Shopify.Multipass.get_url("", customer_data, secret, %{host: "custom-domain.com")
+    "https://custom-domain.com/account/login/multipass/moaqEVx1Yu9hsvYvVpj-LeRYDtOo6ikicfTZd8tR8-xBMRg8tFjGEfllEcjj2VdbsezmT0XuEdglyQzi_biQPkfLJnP1dkxhNtfzwtt6IMQzu3W0qCPzbrUMD_gLaytPVP-zZZuYiSBqEMNdvzFg3zf0TOQHwbizX2D7It02sFI7ZpTRhfX4m_crV0b-DmmF"
+  """
+  @spec get_url(binary, map, binary, map) :: binary
+  def get_url(shop_name, customer_data, multipass_secret, options \\ %{}) do
+    struct(
+      %URI{
+        host: "#{shop_name}.myshopify.com",
+        path: "/account/login/multipass/#{get_token(customer_data, multipass_secret)}",
+        port: 443,
+        scheme: "https"
+      },
+      options
+    )
+    |> URI.to_string()
+  end
+
+  @doc ~S"""
+  Calculates the Shopify Multipass token: an encrypted and signed message containing the customer data to be used for
+  the Multipass SSO.
+  """
+  @spec get_token(map, binary) :: binary
+  def get_token(customer_data, multipass_secret) do
+    {encryption_key, signature_key} = get_keys(multipass_secret)
+
+    Poison.encode!(customer_data)
+    |> encrypt(encryption_key)
+    |> sign(signature_key)
+    |> Base.url_encode64(case: :lower)
+  end
+
+  @doc ~S"""
+  Pads the message string with extra bytes to ensure it is evenly divisible by
+  the block size.
+  See [http://erlang.org/doc/man/crypto.html#block_encrypt-4](http://erlang.org/doc/man/crypto.html#block_encrypt-4)
+  """
+  @spec pad(binary, integer) :: binary
+  def pad(string, block_size \\ @block_size) do
+    to_add = block_size - rem(byte_size(string), block_size)
+    string <> :binary.copy(<<to_add>>, to_add)
+  end
+
+  @doc ~S"""
+  Encrypts a message using the given encryption key. This will pad the message according to the block size.
+  Returns the encrypted message as cipher text.
+  """
+  @spec encrypt(binary, binary, integer) :: binary
+  def encrypt(message, encryption_key, block_size \\ @block_size) do
```

Comment:
In order to make the token testable, we could move the encryption part into a separate module, define a behaviour and implement a test version. That way everyone can either roll with the provided mock encryption, or implement their own - pretty much how the http adapter works.

I am fine with doing that later, or doing it myself and getting this merged as it is though 👍